apps/dashboard/src/components/billing/hooks/use-subscription.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery } from '@tanstack/react-query';
+import { ApiServiceLevelEnum } from '@novu/shared';
+import { get } from '../../../api/api.client';
+import { differenceInDays, isSameDay } from 'date-fns';
+import { useMemo } from 'react';
+
+export interface ISubscriptionData {
+  apiServiceLevel: ApiServiceLevelEnum;
+  isActive: boolean;
+  hasPaymentMethod: boolean;
+  status: string;
+  currentPeriodStart: string | null;
+  currentPeriodEnd: string | null;
+  billingInterval: 'month' | 'year' | null;
+  events: {
+    current: number;
+    included: number;
+  };
+  trial: {
+    isActive: boolean;
+    start: string;
+    end: string;
+    daysTotal: number;
+    daysLeft: number;
+  };
+}
+
+export interface UseSubscriptionType {
+  isLoading: boolean;
+  data: ISubscriptionData | null;
+}
+
+export function useSubscription(): UseSubscriptionType {
```

Comment:
Let's reuse the `@hooks/use-subscription.ts`.

Reply:
Good catch! Didn't noticed we have one